src/rhub/api/monitor/metrics.py

```diff
@@ -117,7 +119,60 @@ def lab_metrics():
 
 
 def bm_metrics():
-    return mock_availability("platforms")
+    # we're going to use mock platform counts for cpu and memory
+    # counts for now since they're not in the BareMetalHost model yet.
+    configs = get_monitoring_configs()
+    available_platforms = configs["mock_platforms"]
+
+    bm_raw_metrics = get_bm_metrics()
+
+    r = []
+
+    for raw_metric in bm_raw_metrics:
+        cpus_used = 0
+        ram_used = 0
+        cpus_available = 0
+        ram_available = 0
+
+        arch = raw_metric["arch"]
+        mock_platform = available_platforms[arch]
+        cpus_per_system = mock_platform["cpus_per_system"]
+        ram_per_system = mock_platform["ram_per_system"]
+
+        r_platform = {}
+        r_platform["platform"] = arch
+
+        status_used = [
+            BareMetalHostStatus.ENROLLING,
+            BareMetalHostStatus.FAILED_ENROLLING,
+            BareMetalHostStatus.RESERVED
+        ]
+
+        status_available = [
+            BareMetalHostStatus.AVAILABLE
+        ]
+
+        for metric, value in raw_metric.items():
+
+            if metric in status_used:
+                cpus_used += (value * cpus_per_system)
+                ram_used += (value * ram_per_system)
+            elif metric in status_available:
+                cpus_available += (value * cpus_per_system)
+                ram_available += (value * ram_per_system)
+
+            if metric != "arch":
+                r_platform[metric] = value
+
+        r_platform["cpus_available"] = cpus_available
+        r_platform["cpus_used"] = cpus_used
+        r_platform["memory_available"] = ram_available
+        r_platform["memory_used"] = ram_used
+
+        r.append(r_platform)
+
+    wrapper = {"data": r}
+    return wrapper
```

Comment:
```suggestion
    raw_data = db.session.query(
        bare_metal_host_full.arch,
        db.func.sum(case(
            (bare_metal_host_full.status.in_(BareMetalHostStatus.host_available_states()), 1),
            else_=0
        )).label('available'),
        db.func.sum(case(
            (bare_metal_host_full.status.in_(BareMetalHostStatus.host_in_use_states()), 1),
            else_=0
        )).label('used'),
    ).group_by(
        bare_metal_host_full.arch,
    ).all()

    data = []
    for arch, available, used in raw_data:
        mock_platform = available_platforms[arch]
        cpus_per_system = mock_platform["cpus_per_system"]
        ram_per_system = mock_platform["ram_per_system"]
        
        data.append(dict(
            platform=arch,
            cpus_available=available * cpus_per_system,
            cpus_used=used * cpus_per_system,
            memory_available=available * ram_per_system,
            memory_used=used * ram_per_system,
        ))

    return {"data": data}
```

I think this can be done this way, if you agree, you won't need the `get_bm_metrics`. And it may facilitate to sum the values once they are on the database.

To use `bare_metal_host_full`, the 'query' code can't be on the model file, it'll generate a circular dependency. I'd suggest to keep it on this file, for now.

Reply:
You mean to keep the query code on the API side? Aren't we crossing boundaries here? (Not that it would necessarily be a bad thing etc.)

I liked the query pretty much but I wonder how we can make this more readable if I need to change this in the future. I think the next iteraction (when nodes' RAM, storage and CPUs are known) should use a PostgreSQL view (for speed / maintainability.)

Reply:
> You mean to keep the query code on the API side?

Yes, keep it on `src/rhub/api/monitor/metrics.py` or create a file `src/rhub/api/bare_metal/util.py` with this query.

> Aren't we crossing boundaries here?

There's already query code on many (or all) endpoint implementation.

> I liked the query pretty much but I wonder how we can make this more readable if I need to change this in the future.

We'll need to think, but it has fewer code lines to read... haha

> should use a PostgreSQL view

I'm not sure if our frameworks provide support for this. And if so, I'm not sure how much RHub code needs to be database agnostic.

Reply:
In terms of views it should just query the thing as a regular table. But for now this suffices.